### ISSUE TITLE
Regex support for menu aliases

### ISSUE THREAD

**Issue by Victor-Bo** [ventoy/Ventoy#579]
I use Ventoy for a long time now and a functionnality I always missed would be a support for wildcards or even whole regex. It would be better to be able to make aliases for `manjaro-xfce-*.iso` instead of `manjaro-xfce-20.1.2-201019-linux58.iso`

Because now if you download a new version of Manjaro in this case you need to update the config with the exact name which is very redundant when you put new ISOs.

Another functionnality would be capture groups for example
```
{
  "image": "manjaro-(.+?)-(.+?)-.*.iso"
  "key": "Manjaro \1 \2"
}
```
would transform `manjaro-xfce-20.1.2-201019-linux58.iso` into `Manjaro xfce 20.1.2`

**Comment by ventoy** [comment]
Well, I think menu alias is only something about the menu display name. It does not affect main function. 
And menu alias is a very lightweight plugin, so regex match will be too complex and will reduce the matching efficiency.

**Comment by Victor-Bo** [comment]
Yes I understand but would It be really that slow? Wouldn't it be valuable to have a bit more of latency, having a functionnality I think everyone would use and that is incredibly convenient to have?

**Comment by aaronhatesregex** [comment]
> And menu alias is a very lightweight plugin, so regex match will be too complex and will reduce the matching efficiency.

Hi, if not RegEx (I hate that as you can tell from my username), would you please cousider wildcard support?

A minimum support for most common wildcards, i.e. asterisk (*) and question mark (?), would make a difference.

I have a script to update `ventoy.json` but it just seems kinda weird.

**Comment by ventoy** [comment]
You can test with this CI release:
https://github.com/ventoy/Ventoy/actions/runs/780095502

I add a simple * match feature.
But for simplicity, the * fuzzy matching is much different with our common usage.
```
A * is fixed to match only 1 character, if you want to match multiple characters, you need to write multiple *
```

Sample :
```
{
    "menu_alias" : [
        {
            "image": "/Ubuntu_based/KDEneon/neon-user-********.iso",
            "alias": "KDEneon - Stable"
        }
    ]
}
```

**Comment by ifohancroft** [comment]
I'd also love to be able to match multiple characters with a single '*' character, or even have regex support.

Right now Parrot OS's version for example is 5.3, but who knows what their version number policy is, maybe it would reach version 5.10 then we'll need to change the config file.

Same problem with Windows. The current Windows 11 ISO is called Win11_22H2_English_x64v2, but what if the version (22H2) gets longer or shorter, or they drop the v2 at the end?

**Comment by victorbnl** [comment]
Instead of regexes and multi-character wildcards, maybe a prefix detection could suffice? If an ISO starts with `Win`, or `Win10` for more precision, we know what OS it is going to be. If it starts with `ubuntu` as well.

**Comment by tefod-zz** [comment]
> Instead of regexes and multi-character wildcards, maybe a prefix detection could suffice? If an ISO starts with `Win`, or `Win10` for more precision, we know what OS it’ll turn out to be. If it starts with `ubuntu` as well. In terms of performance, iterating over a few characters doesn’t look very resource-intensive.

Functionally, it would be the same, and we would be very happy about such an update.
This way, we wouldn't have to adjust the ISO filenames or change the menu entries with each new version of the ISO.

**Comment by ehawman** [comment]
I would really love to use regex to match strings. That or glob instead of single-character wildcard. Naming conventions change with the wind, and keeping the image names updated is tedious.
